### PR TITLE
KEB: next-gen runtimes endpoint

### PR DESCRIPTION
**Pull request by ebensom**
**Description**

These changes are needed for Kyma 2.0 central monitoring dependency on runtimes state and component configuration used by the AVS integration.

More details:
https://github.tools.sap/kyma/backlog/issues/1451

Changes proposed in this pull request:

- Operation detail parameter can specify whether runtimes endpoint shall return all operations, or only the last operations per runtime. This can have a huge performance benefit when querying all runtimes periodically from a service which doesn't need all operation details. 
- Add runtime state field into `RuntimeDTO.Status.State`, compute runtime state based on last operation state.
- Expose latest kyma config and cluster config optionally per runtime when requested using `kyma_config` and `cluster_config` parameters. 
- Expose AVS internal evaluation ID as a new field.
- Allow `runtime.NewClient` to accept any HTTP client to use underlying API calls

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/runtime/handler.go [hunk 1]
```diff
@@ -148,6 +123,157 @@ func (h *Handler) takeLastNonDryRunClusterOperations(oprs []internal.UpgradeClus
...
+			dto.Status.State = pkg.StateError
+		}
+	case string(domain.InProgress):
+		switch lastOp.Type {
+		case pkg.Provision, pkg.Unsuspension:
+			dto.Status.State = pkg.StateProvisioning
+		case pkg.Deprovision, pkg.Suspension:
+			dto.Status.State = pkg.StateDeprovisioning
+		case pkg.UpgradeKyma, pkg.UpgradeCluster:
+			dto.Status.State = pkg.StateUpgrading
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/runtime/handler.go:147]: let's add here an Upgrade type also (the PR should be merged soon, see update process  https://github.com/kyma-project/control-plane/pull/795)

File: components/kyma-environment-broker/common/runtime/client.go [hunk 2]
```diff
@@ -29,13 +27,12 @@ type client struct {
 
 // NewClient constructs and returns new Client for KEB /runtimes API
 // It takes the following arguments:
-//   - ctx  : context in which the http request will be executed
-//   - url  : base url of all KEB APIs, e.g. https://kyma-env-broker.kyma.local
-//   - auth : TokenSource object which provides the ID token for the HTTP request
-func NewClient(ctx context.Context, url string, auth oauth2.TokenSource) Client {
+//   - url        : base url of all KEB APIs, e.g. https://kyma-env-broker.kyma.local
+//   - httpClient : underlying HTTP client used for API call to KEB
+func NewClient(url string, httpClient *http.Client) Client {
```
Comment (ksputo) [inline components/kyma-environment-broker/common/runtime/client.go:32]: I see that implementation of the KEB client changed a bit, but you didn't introduce importing new version here:
https://github.com/kyma-project/control-plane/blob/294a1afd8f073957c80cc637681b406128ffbd38/tools/cli/pkg/command/taskrun.go#L207-L207

Are you planning to add it in separate PR after this one is merged?

[the remark this answers is not in the document]
Comment (ebensom) [reply components/kyma-environment-broker/common/runtime/client.go:32]: Yes, I plan to make KCP CLI change in a separate PR after this one.

**Review by ksputo — APPROVED** [review]
lgtm

**Pull request merged**